### PR TITLE
Acm.md comments

### PR DESCRIPTION
**Pull request by MichaelJr1**
Added comments in example code so future students understand each line of code
Fixes#107

### CODE REVIEW

**Review by onzfonz — CHANGES_REQUESTED** [review]
Please put the note about not making lengthy comments outside of the java file as actual text.
Do not use all caps there, but use the formatting for markdown to help with this.

I would also reword the comments for init run and main, by telling them that they really only should be changing run and init should only be about changings the sizes and main they should leave alone but that is the start of the program.

Once you make those changes we'll have one more round of changes

**Comment by MichaelJr1** [comment]
Made the changes, let me know if there is anything else that I should change

File: docs/labs/4-ACM.md [hunk 1]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
-import acm.graphics.*;
-import acm.program.*;
-import acm.util.*;
-import java.awt.*;
 
+// The next 4 lines are imported necessary libraries for graphics and utilities
+import acm.graphics.*; // This imports classes for drawing graphics
+import acm.program.*; // This imports classes for creating graphical programs
+import acm.util.*; // This imports classes for utility programs
+import java.awt.*; // // This imports classes for Abstract Window Toolkit (AWT) for graphics
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:55]:
```suggestion
// These 4 lines import necessary libraries for graphics and utilities
import acm.graphics.*; // imports classes for drawing graphics
import acm.program.*;  // imports classes for creating graphical programs
import acm.util.*;     // imports classes made to make writing programs easier
import java.awt.*;     // imports classes for Abstract Window Toolkit (AWT) for graphics
```

File: docs/labs/4-ACM.md [hunk 2]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
+
+// This defines a class named 'ACMLab' and inherits it from another class named 'GraphicsProgram'
+// These are the constants for the program window dimensions
 public class ACMLab extends GraphicsProgram {
-    public static final int PROGRAM_HEIGHT = 600;
-    public static final int PROGRAM_WIDTH = 800;
+    public static final int PROGRAM_HEIGHT = 600; // This sets the Height size of the program window
+    public static final int PROGRAM_WIDTH = 800; // This sets the Width size of the program window
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:61]:
```suggestion
    public static final int PROGRAM_HEIGHT = 600; // Sets the Height of the program window in pixels
    public static final int PROGRAM_WIDTH = 800; // does the same for the Width
```

File: docs/labs/4-ACM.md [hunk 3]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
 
+    // The initialization method 'init' is about changing the sizes
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:63]:
```suggestion
    // The method 'init' should not be changed; it is only for setting the window size
```

File: docs/labs/4-ACM.md [hunk 4]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
     public void init() {
-        setSize(PROGRAM_WIDTH, PROGRAM_HEIGHT);
-        requestFocus();
+        setSize(PROGRAM_WIDTH, PROGRAM_HEIGHT); // This sets the size of the program window to predefined constants
+        requestFocus(); // This ensures the program window receives input focus
     }
 
+    // This part of the program is about adding GLabels to the program
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:69]:
```suggestion
    // run is your starting point for your program; the code here adds GLabels to the program
```

File: docs/labs/4-ACM.md [hunk 5]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
     public void run() {
-        add(new GLabel("Hello World", 300, 300));
-    }
+        // This adds a GLabel that displays "Hello World" at coordinates (300, 300)
+        add(new GLabel("Hello World", 300, 300)); // The coordinates can be altered to change label placement
 
+    // This is the main method that starts the program
+    // It is recommended to leave this part of the program alone
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:75]:
```suggestion
    // To abide by java's rules, we have the main method that starts the program
    // Leave this part alone; start is another method that calls init and run.
```

File: docs/labs/4-ACM.md [hunk 6]
```diff
@@ -43,27 +43,39 @@ open **ACMLab.java** in the editor.
...
     public static void main(String[] args) {
-        new ACMLab().start();
+        //This creates a new instance of the ACMLab class, it then calls its start method to begin execution
```
Comment (onzfonz) [inline docs/labs/4-ACM.md:77]:
```suggestion
        // Because it is static, it creates a new instance of the ACMLab class, then calls start on it
```

**Review by onzfonz — CHANGES_REQUESTED** [review]
Please use add suggestions to batch from the files changed tab.
Once this happens you may have one more round to request changes and then I would approve
make sure you use this button
image

you can access it from the files changed tab in the pull request.

**Review by onzfonz — APPROVED** [review]
LGTM

**Comment by onzfonz** [comment]
@MichaelJr1 approved, all that is left is for you to update the branch and merge via the green button yourself

**Pull request merged**